src/WebJobs.Script/Binding/FunctionBinding.cs

```diff
@@ -196,8 +197,10 @@ internal static async Task BindAsyncCollectorAsync<T>(BindingContext context)
                     throw new ArgumentException("Unsupported collection type.");
                 }
 
-                await collector.AddAsync((T)converted);
+                tasks.Add(collector.AddAsync((T)converted));
```

Comment:
I think parallelizing this will cause problems for some existing IAsyncCollector implementations. For example, look at the CosmosDB collector [here](https://github.com/Azure/azure-webjobs-sdk-extensions/blob/a0fb2011acfa88126d9eecc285d7a7f8283ccbe5/src/WebJobs.Extensions.CosmosDB/Bindings/CosmosDBAsyncCollector.cs#L28). It's doing an optimistic upsert, and if the DB or collection doesn't exist, it'll handle the error, create the collection and try the upsert again. If all the Adds were parallelized, they will likely always fail into the Create DB/Collection case and result in a bunch of retries/failures. @ealsur what are your thoughts here? Likely there are other existing collectors that will have similar issues.

I think rather than making this change, extensions should just implement IAsyncCollector.Flush batch support, so AddAsync just accumulates, and can perform efficient batched service operations.

Reply:
Also, if the customer wants parallelization, they can parallelize the AddAsync on the IAsyncCollector, doing the same WhenAll pattern.